packages/providers/src.ts/base-provider.ts

```diff
@@ -1552,20 +1552,9 @@ export class BaseProvider extends Provider implements EnsProvider {
     async getTransaction(txId: string | Promise<string>): Promise<TransactionResponse> {
         await this.getNetwork();
         txId = await txId;
-        const [ accId, , ] = txId.split("-");
-        const ep = '/api/v1/transactions?account.id=' + accId;
+        const ep = '/api/v1/transactions/'+txId;
         let { data } = await axios.get(this.mirrorNodeUrl + ep);
-        while (data.links.next != null) {
-            const filtered = data.transactions
-                .filter((e: HederaTxRecordLike) =>
-                    e.transaction_id.toString() === txId && e.result === 'SUCCESS');
-
-            if (filtered.length > 0) {
-                return filtered[0];
-            }
-            ({ data } = await axios.get(this.mirrorNodeUrl + data.links.next));
-        }
-        return null;
+        return data.transactions.length > 0 ? data.transactions[0] : null;
```

Comment:
In some cases there may be duplicate transactions with the same TX ID.
In that case we might have `n` number of transactions where only one will have status SUCCESS and all others will have DUPLICATED.

We will need to filter the transactions result and search for the `SUCCESS`. Once we have it, we should return that and not the `0` index

Reply:
Will handle it.